### PR TITLE
Update proto package to version 4, resolve name conflict

### PR DESCRIPTION
**Pull request by catsby**
**TLDR:** 
Update the protocol buffer version to 4 and rename the `types.proto` file to `kvtypes.proto` to avoid name registration conflict.

------

The [hashicorp/vault][vault] project currently shows an error with every `vault` command:

```
[vault][master](4)$ vault help 
2020/05/21 07:42:59 WARNING: proto: file "types.proto" is already registered
A future release will panic on registration conflicts. See:
https://developers.google.com/protocol-buffers/docs/reference/go/faq#namespace-conflict
```

While tracking that down I landed here where we're using an older version of protocol buffers:

https://github.com/hashicorp/vault-plugin-secrets-kv/blob/72880f3e87bc4e35ac0654d48fd3e6450d96f790/types.pb.go#L23

I updated the protocol version by adjusting/adding the `go_package` and `package` statements in `types.go`, and ran `make proto` to regenerate the files (I had to update the `makefile` as well, see included change). Rebuilding Vault with the updated kv dependency then gave this error:

```
2020/05/21 07:49:00 WARNING: proto: file "types.proto" is already registered
        previously from: "github.com/hashicorp/go-kms-wrapping"
        currently from:  "github.com/hashicorp/vault-plugin-secrets-kv"
A future release will panic on registration conflicts. See:
https://developers.google.com/protocol-buffers/docs/reference/go/faq#namespace-conflict
```

In this PR we update the protocol version and rename `types.proto` to `kvtypes.proto` (and corresponding `*.pb.go` files) to avoid the name conflict.

Originally I attempted to put `types.proto` into a sub directory and into it's own package like we do in [go-raftchunking][raftchunk], but we add methods to the `Configuration` type which wont work if it's a separate package:

- https://github.com/hashicorp/vault-plugin-secrets-kv/blob/72880f3e87bc4e35ac0654d48fd3e6450d96f790/delete_version_after.go#L46-L64

This should resolve type name conflict in the main Vault repository once Vault updates. Note that I considered trying to update/change `go-kms-wrapping` to use the sub directory approach but it's already on version 4 of protocol buffers so I thought the changes are best to start here where we're in need of updating anyway. 

[vault]: https://github.com/hashicorp/vault
[raftchunk]: https://github.com/hashicorp/go-raftchunking/blob/7e9e8525653a30aa1ad9cbed9ebc4a814ffe818c/types/types.proto

### CODE REVIEW

**Comment by jefferai** [comment]
The file in go-kms-wrapping has been changed but you should still change it here as `types.proto` is used in various places in Vault too. Although I don't know if it takes directory paths into account.
